VERSIONING.md

```diff
@@ -0,0 +1,71 @@
+# NUnit Console Runner and Engine Versioning
+
+Both **Console** and **Engine** packages use a version number in the form MAJOR.MINOR.PATCH. We change the
+
+- MAJOR version when we have made incompatible API changes
+- MINOR version when adding functionality in a backwards-compatible manner
+- PATCH version when you making backwards-compatible bug fixes
+
+This versioning approach is, of course, what is known as "Semantic Versioning." Taken by itself, however, the above description leaves out some rather important stuff...
+
+* What exactly are the APIs we are protecting from incompatible changes?
+* What degree of change is needed in each of those APIs for it to be considered a breaking change?
+
+The remainder of this document attempts to answer those questions.
+
+## Console Runner API
+
+First, let's note that the **Console Runner** is not designed to be used as a library, so there is no ABI or library-like API to worry about. However, the runner does present a command-line interface, which is used both interactively and through user-created scripts. The command-line API is defined by our documentation, including the help message the runner displays and the [published documentation](https://docs.nunit.org/articles/nunit/running-tests/Console-Command-Line.html) on our website.
+
+### Examples of MAJOR Command-Line Changes
+
+- Complete removal of an option.
+- Changing the behavior of an option so that it no longer does what it was defined to do.
+- Removing the ability to run under a particular platform.
+
+### Examples of MINOR Command-Line Changes
+
+- Addition of a new option
+- Modifying the behavior of an option so it does the same thing better or more effectively or has added sub-options.
+- Adding support for running under an new platform
+
+### Examples of PATCH Command-Line changes
+
+- Fixing a bug in the behavior of an option.
+- Restoring an option to its original defined behavior.
```

Comment:
How does this differ from 'fixing a bug'?

Reply:
I combined the two. I wanted to be clear that the __defined__ behavior is what counts, not the actual (broken) behavior. In practice, of course, we would probably be less rigid about this if the bad behavior had gone unnoticed over multiple releases.